plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -236,45 +234,46 @@ func (p *HelmChartInflationGeneratorPlugin) Generate() (resmap.ResMap, error) {
 	if idx := strings.Index(stdoutStr, "---"); idx != -1 {
 		return p.h.ResmapFactory().NewResMapFromBytes([]byte(stdoutStr[idx:]))
 	}
-	return nil, rmfErr
+	return nil, err
 }
 
-func (p *HelmChartInflationGeneratorPlugin) getTemplateCommandArgs() []string {
+func (p *HelmChartInflationGeneratorPlugin) templateCommand() []string {
 	args := []string{"template"}
 	if p.ReleaseName != "" {
 		args = append(args, p.ReleaseName)
 	}
-	args = append(args, filepath.Join(p.ChartHome, p.ChartName))
-	if p.ReleaseNamespace != "" {
-		args = append(args, "--namespace", p.ReleaseNamespace)
+	args = append(args, filepath.Join(p.absChartHome(), p.Name))
+	if p.ValuesFile != "" {
+		args = append(args, "--values", p.ValuesFile)
 	}
-	if p.Values != "" {
-		args = append(args, "--values", p.Values)
+	if p.ReleaseName == "" {
+		// AFAICT, this doesn't work as intended due to a bug in helm.
+		// See https://github.com/helm/helm/issues/6019
+		// I've tried placing the flag before and after the name argument.
+		args = append(args, "--generate-name")
 	}
-	args = append(args, p.ExtraArgs...)
 	return args
 }
 
-func (p *HelmChartInflationGeneratorPlugin) getPullCommandArgs() []string {
-	args := []string{"pull", "--untar", "--untardir", p.ChartHome}
-	chartName := fmt.Sprintf("%s/%s", p.ChartRepoName, p.ChartName)
-	if p.ChartVersion != "" {
-		args = append(args, "--version", p.ChartVersion)
-	}
-	if p.ChartRepoURL != "" {
-		args = append(args, "--repo", p.ChartRepoURL)
-		chartName = p.ChartName
+func (p *HelmChartInflationGeneratorPlugin) pullCommand() []string {
+	args := []string{"pull", "--untar", "--untardir", p.absChartHome()}
+	if p.Version != "" {
+		args = append(args, "--version", p.Version)
+	}
+	chartName := p.Name
+	if p.Repo != "" {
+		args = append(args, "--repo", p.Repo)
+	} else {
+		// This doesn't make sense.
+		chartName = filepath.Join(p.ChartHome, p.Name)
```

Comment:
Should the comment include an explanation for why this doesn't make sense?

Reply:
will do (actually will drop)

Reply:
done